src/js_results.hpp

```diff
@@ -160,17 +160,22 @@ inline void setup_aliases(parser::KeyPathMapping &mapping, const realm::SharedRe
 {
     const realm::Schema &schema = realm->schema();
     for (auto it = schema.begin(); it != schema.end(); ++it) {
+        TableRef table;
         for (const Property &property : it->computed_properties) {
             if (property.type == realm::PropertyType::LinkingObjects) {
+                if (!table) {
```

Comment:
Any reason this is still inside the loop of properties and not just defined where you define the `table` variable? It doesn't seem to change for each property?

Reply:
`ObjectStore::table_for_object_type()` is an expensive call, and I only want to do it when needed (linking object property or a persisted property. The idea is to call it at most one time.

Reply:
But we don't support schemas with no properties, right? So having the code twice just seems to reduce readability for no real benefit?

Reply:
I see your point. Even in the case where a schema doesn't have any properties, we could do the call and avoid the branching. I mean, it is likely to be very rare that a schema doesn't have a property.